### PR TITLE
Add a CodeLens API sample

### PR DESCRIPTION
**Pull request by fcrespo82**
# CodeLens Sample

This is a sample extension that shows the usage of the CodeLens API.

It is not intended as a production quality extension.

- Create a new file
- Write anything
- Click on the CodeLens for action example
- Can be enabled or disabled by command palete

# How it works, what it shows?

- The extension uses the [`CodeLens`](https://code.visualstudio.com/docs/extensionAPI/vscode-api#CodeLens)

# How to run locally

* open this folder in VS Code and press `F5`

### CODE REVIEW

**Comment by panlina** [comment]
@octref Why not merge this?

**Comment by Tyriar** [comment]
Ping @octref, please assign to someone else if you're not going to take this.

File: codelens-sample/package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    "default": true
+                }
+            }
+        }
+    },
+    "scripts": {
+        "vscode:prepublish": "npm run compile",
+        "compile": "tsc -p ./",
+        "watch": "tsc -watch -p ./",
+        "postinstall": "node ./node_modules/vscode/bin/install"
```
Comment (octref) [inline codelens-sample/package.json]: Please move to `@types/vscode`, see https://github.com/microsoft/vscode-extension-samples/blob/master/helloworld-sample/package.json

File: codelens-sample/src/CodelensProvider.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import * as vscode from 'vscode';
+/**
```
Comment (octref) [inline codelens-sample/src/CodelensProvider.ts]: Spacing here

File: codelens-sample/src/extension.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+'use strict';
```
Comment (octref) [inline codelens-sample/src/extension.ts]: No need for this

File: codelens-sample/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        ],
+        "sourceMap": true,
+        "rootDir": "src",
+        /* Strict Type-Checking Option */
+        "strict": true,   /* enable all strict type-checking options */
+        /* Additional Checks */
+        "noUnusedLocals": true /* Report errors on unused locals. */
+        // "noImplicitReturns": true, /* Report error when not all code paths in function return a value. */
+        // "noFallthroughCasesInSwitch": true, /* Report errors for fallthrough cases in switch statement. */
+        // "noUnusedParameters": true,  /* Report errors on unused parameters. */
```
Comment (octref) [inline codelens-sample/tsconfig.json]: You can removed the commented out ones.

File: codelens-sample/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+- Click on the CodeLens for action example
+- Can be enabled or disabled by command palete
+
+# How it works, what it shows?
+
+- The extension uses the [`CodeLens`](https://code.visualstudio.com/docs/extensionAPI/vscode-api#CodeLens)
+
+# How to run locally
+
+* open this folder in VS Code and press `F5`
```
Comment (octref) [inline codelens-sample/README.md]: Please model it after the helloworld-sample by:

- Add a gif that demonstrates the usage
- Adding a "VS Code API" section
- Adding a "Running the Sample" section

https://github.com/microsoft/vscode-extension-samples/tree/master/helloworld-sample#vs-code-api

**Comment by fcrespo82** [comment]
@octref  All should be resolved.

File: .scripts/samples.js [hunk 6]
```diff
@@ -177,6 +177,13 @@ const samples = [
     guide: '/api/extension-guides/custom-data-extension',
     apis: [],
     contributions: []
+  },
+  {
+    description: 'CodeLens Sample',
+    path: 'codelens-sample',
+    guide: null,
+    apis: [`languages.registerCodeLensProvider`, `CodelensProvider`, `CodeLens`],
```
Comment (KamasamaK) [inline .scripts/samples.js]: Should be `CodeLensProvider`

Comment (fcrespo82) [reply .scripts/samples.js]: done

Comment (KamasamaK) [reply .scripts/samples.js]: Sorry. If it wasn't clear, this was referring to line 185. Specifically, the change from `CodelensProvider` to `CodeLensProvider` -- `l` to `L`.

Comment (fcrespo82) [reply .scripts/samples.js]: Oh, I see. Should I revert back the change on line 182?

**Comment by octref** [comment]
LGTM, thanks!

**Pull request merged**